client/cache_client/cache_client.go

```diff
@@ -257,7 +276,7 @@ func (c *ClientWrapper) Put(key string, value string) error {
 	// check response
 	res, err := new(http.Client).Do(req)
 	defer res.Body.Close()
```

Comment:
I think this line is causing the test to fail with a segfault. The error check needs to go before using the res object

Reply:
Wokay, done pushing the fix.